sw/device/tests/aes_force_prng_reseed_test.c

```diff
@@ -0,0 +1,264 @@
+
+  // T3:  Wait for status kDifAesStatusIdle
+  // Start AES with the given transaction
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusIdle, true, kTestTimeout);
+
+  // T4: dif_aes_start
+  CHECK_DIF_OK(dif_aes_start(&aes, &transaction, &key, NULL));
+
+  LOG_INFO("Disabling CSRNG...");
+  CHECK_DIF_OK(dif_csrng_stop(&csrng));
+  CHECK_DIF_OK(dif_entropy_src_stop(&entropy_src));
+
+  // Re-enable Entropy Source and CSRNG in the proper sequence
+  LOG_INFO("Re-initializing Entropy Source and CSRNG...");
+
+  // Initialize and configure Entropy Source
+  dif_entropy_src_config_t entropy_src_config =
+      entropy_testutils_config_default();
+  CHECK_DIF_OK(dif_entropy_src_configure(&entropy_src, entropy_src_config,
+                                         kDifToggleEnabled));
+
+  // Initialize and start CSRNG
+  CHECK_DIF_OK(dif_csrng_configure(&csrng));
+  // set_edn_auto_mode();
+  LOG_INFO("Setting EDN to auto mode...");
+  TRY(entropy_testutils_auto_mode_init());
+  // CHECK_DIF_OK(entropy_testutils_auto_mode_init());
+  monitor_edn_status(&edn0, &edn1);
+
+  // Trigger PRNG reseed request
+  LOG_INFO("Triggering PRNG reseed...");
+  CHECK_DIF_OK(dif_aes_trigger(&aes, kDifAesTriggerPrngReseed));
+
+  // T5: "Convert" plain data byte arrays to `dif_aes_data_t`
+  dif_aes_data_t in_data_plain;
+  memcpy(in_data_plain.data, kAesModesPlainText, sizeof(in_data_plain.data));
+
+  // T6: Wait ofr Status kDifAesStatusIdle
+  LOG_INFO(
+      "Now Starting AES encryption process firstly by observing if AES is in "
+      "Idle state");
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusIdle, true, kTestTimeout);
+
+  // T7: Wait for Status input DifAesStatusInputReady
+  LOG_INFO("AES started: ECB encryption, waiting for input ready.");
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusInputReady, true,
+                                kTestTimeout);
+  // T8: Load the plain text to trigger the encryption operation
+  CHECK_DIF_OK(dif_aes_load_data(&aes, in_data_plain));
+
+  // Trigger AES reseed request, specific for manual operation case
+  LOG_INFO("AES encryption started.");
+  CHECK_DIF_OK(dif_aes_trigger(&aes, kDifAesTriggerStart));
+
+  busy_spin_micros(500);
+
+  LOG_INFO("Disabling CSRNG...");
+  CHECK_DIF_OK(dif_csrng_stop(&csrng));
+  CHECK_DIF_OK(dif_entropy_src_stop(&entropy_src));
+
+  // Re-enable Entropy Source and CSRNG in the proper sequence
+  LOG_INFO("Re-initializing Entropy Source and CSRNG...");
+  // Initialize and configure Entropy Source
+  dif_entropy_src_config_t entropy_src_config_1 =
+      entropy_testutils_config_default();
+  CHECK_DIF_OK(dif_entropy_src_configure(&entropy_src, entropy_src_config_1,
+                                         kDifToggleEnabled));
+  // Initialize and start CSRNG
+  CHECK_DIF_OK(dif_csrng_configure(&csrng));
+
+  LOG_INFO("Setting EDN to auto mode...");
+  TRY(entropy_testutils_auto_mode_init());
+  // CHECK_DIF_OK(entropy_testutils_auto_mode_init());
```

Comment:
It is not clear to me why we are doing this?

Reply:
Noted.. 
Removing the redundant logic...
We already enabled entropy complex earlier...